rust/tests/command_filesystem_check.rs

```diff
@@ -0,0 +1,112 @@
+#![cfg(all(feature = "integration_test"))]
+
+use deltalake::test_utils::{
+    set_env_if_not_set, IntegrationContext, StorageIntegration, TestResult, TestTables,
+};
+use deltalake::DeltaOps;
+use deltalake::Path;
+use serial_test::serial;
+
+mod common;
+
+#[tokio::test]
+#[serial]
+async fn test_filesystem_check_local() -> TestResult {
+    Ok(test_filesystem_check(StorageIntegration::Local).await?)
+}
+
+#[cfg(any(feature = "s3", feature = "s3-rustls"))]
+#[tokio::test]
+#[serial]
+async fn test_filesystem_check_aws() -> TestResult {
+    set_env_if_not_set("AWS_S3_ALLOW_UNSAFE_RENAME", "true");
+    set_env_if_not_set("AWS_S3_LOCKING_PROVIDER", "none");
```

Comment:
Could be wrong, but I *think* you only need the first environment variable. If that's not the case, we might want to file an issue for that.
```suggestion
```

Reply:
Yes it's required. I had a commit  with just allow unsafe set and it failed since http is not supported for the lock client. Currently the lock client is always constructed and is not determined by `UNSAFE_RENAME`

See: https://github.com/delta-io/delta-rs/blob/main/rust/src/storage/s3.rs#L402